### PR TITLE
Update MXNet to 1.5.0

### PR DESCRIPTION
**Pull request by hqucms**
This is based on the official [MXNet 1.5.0](https://github.com/apache/incubator-mxnet/releases/tag/1.5.0), with the thread-safety patch (introduced in https://github.com/cms-sw/cmsdist/pull/4207) applied for use in CMSSW. The new version is needed for some ongoing developments for a (possible) future version of DeepAK8. 

The code with the patch applied currently lives in [my repo](https://github.com/hqucms/incubator-mxnet/tree/1.5.0-cms-mod) and needs to be ported to [cms-externals](https://github.com/cms-externals/incubator-mxnet/).

### CODE REVIEW

**Comment by mrodozov** [comment]
please test
lets try it I'll move it to externals later

**Comment by mrodozov** [comment]
the branch is pushed https://github.com/cms-externals/incubator-mxnet/tree/1.5.0-cms-mod

**Comment by mrodozov** [comment]
please test

File: mxnet-predict-toolfile.spec [hunk 1]
```diff
@@ -9,13 +9,14 @@ Requires: mxnet-predict
...
-  <lib name="mxnetpredict"/>
+  <lib name="mxnet"/>
   <client>
     <environment name="MXNET_PREDICT_BASE" default="@TOOL_ROOT@"/>
     <environment name="INCLUDE" default="$MXNET_PREDICT_BASE/include"/>
-    <environment name="LIBDIR" default="$MXNET_PREDICT_BASE/lib"/>
+    <environment name="LIBDIR" default="$MXNET_PREDICT_BASE/lib64"/>
   </client>
   <use name="openblas"/>
+  <use name="lapack"/>
```
Comment (davidlange6) [inline mxnet-predict-toolfile.spec]: hi- why is this needed? better to use openblas instead.

Comment (hqucms) [reply mxnet-predict-toolfile.spec]: @davidlange6 The CMake build fails without providing the standalone LAPACK library. I put a workaround in https://github.com/cms-sw/cmsdist/pull/5148/commits/03a8093b04eba91a5dab1a80fccb726a3d411dfc.

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
+externals

**Comment by mrodozov** [comment]
@davidlange6 any concern about merging this ?
